warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -2072,15 +2072,26 @@ public ASTJexlScript addDateFilters(final ASTJexlScript queryTree, ScannerFactor
             JexlNode andNode = JexlNodeFactory.createAndNode(andChildren);
             JexlNodeFactory.setChildren(queryTree, Collections.singleton(andNode));
 
-            // now lets update the query parameters with the correct start and
-            // end dates
-            log.info("Remapped " + dateType + " dates [" + config.getBeginDate() + "," + config.getEndDate() + "] to EVENT dates "
-                            + dateIndexData.getBeginDate() + "," + dateIndexData.getEndDate());
-
-            // reset the dates in the configuration, no need to reset then in
-            // the Query settings object
-            config.setBeginDate(dateIndexData.getBeginDate());
-            config.setEndDate(dateIndexData.getEndDate());
+            if (config.getNoExpansionIfCurrentDateTypes().contains(dateType) && dateIndexData.getBeginDate().before(config.getBeginDate())) {
+                // now lets update the query parameters with the correct start and
+                // end dates
+                log.info("Remapped " + dateType + " dates [" + config.getBeginDate() + "," + config.getEndDate() + "] to EVENT dates " + config.getBeginDate()
+                                + "," + dateIndexData.getEndDate());
+
+                // reset the dates in the configuration, no need to reset them in
+                // the Query settings object
+                config.setEndDate(dateIndexData.getEndDate());
+            } else {
+                // now lets update the query parameters with the correct start and
+                // end dates
+                log.info("Remapped " + dateType + " dates [" + config.getBeginDate() + "," + config.getEndDate() + "] to EVENT dates "
+                                + dateIndexData.getBeginDate() + "," + dateIndexData.getEndDate());
+
+                // reset the dates in the configuration, no need to reset them in
+                // the Query settings object
+                config.setBeginDate(dateIndexData.getBeginDate());
+                config.setEndDate(dateIndexData.getEndDate());
+            }
```

Comment:
The query planner used to blindly set the begin and end date from the date index data, and now we're respecting the user-supplied behind date if the date type matches the no expansion thing.

Should the query planner also respect the user-supplied end date?

Reply:
This is the query planner.  Setting the dates in the configuration serve that purpose.  Am I missing the question here?

Reply:
ok, readjusted to avoid updating begin date altogether if our special type, and to only update end date if not current date.